### PR TITLE
chore(dp): single-step integration tests

### PR DESCRIPTION
**Pull request by WojSad**
Signed-off-by: Wojciech Sadowy <wojciech.sadowy@freedomfi.com>

chore(dp): single-step integration tests

## Summary

Refactored existing integration tests for dp with orc8r. Added new tests including single-step related logic.

## Test Plan

Tests included

### CODE REVIEW

File: dp/cloud/go/services/dp/storage/cbsd_manager.go [hunk 1]
```diff
@@ -16,11 +16,11 @@ package storage
 import (
 	"database/sql"
 
-	sq "github.com/Masterminds/squirrel"
-
 	"magma/dp/cloud/go/services/dp/storage/db"
 	"magma/orc8r/cloud/go/sqorc"
 	"magma/orc8r/lib/go/merrors"
+
+	sq "github.com/Masterminds/squirrel"
```
Comment (jkmar) [inline dp/cloud/go/services/dp/storage/cbsd_manager.go]: `go imports`

File: dp/cloud/go/services/dp/storage/cbsd_manager.go [hunk 2]
```diff
@@ -240,11 +240,15 @@ func (c *cbsdManagerInTransaction) updateCbsd(networkId string, id int64, data *
...
 func (c *cbsdManagerInTransaction) enodebdUpdateCbsd(data *DBCbsd) error {
-	mask := db.NewIncludeMask("id")
+	mask := db.NewIncludeMask(getEnodebdWritableFields()...)
 	filters := sq.Eq{"cbsd_serial_number": data.CbsdSerialNumber}
-	if _, err := c.selectForUpdateIfCbsdExists(mask, filters); err != nil {
+	cbsd, err := c.selectForUpdateIfCbsdExists(mask, filters)
+	if err != nil {
 		return err
 	}
+	if !ShouldENodeBDUpdate(cbsd, data) {
```
Comment (jkmar) [inline dp/cloud/go/services/dp/storage/cbsd_manager.go:249]: Where is this function defined (I think you forgot to add some file to git)

File: orc8r/cloud/go/sqorc/error_checker_test.go [hunk 3]
```diff
@@ -78,29 +78,31 @@ func (s *ErrorCheckerTestSuite) TestErrorCheckerDefaultsToPostgres() {
...
 func (s *ErrorCheckerTestSuite) TestSQLiteGetError() {
-	testCases := []sqliteGetErrorTestCase{
-		{
-			name:    "test sqlite constraint error with SQLiteErrorChecker",
-			checker: SQLiteErrorChecker{},
-			err: sqlite3.Error{
-				Code: sqlite3.ErrConstraint,
-			},
-			expectedError: merrors.ErrAlreadyExists,
+	testCases := []sqliteGetErrorTestCase{{
```
Comment (jkmar) [inline orc8r/cloud/go/sqorc/error_checker_test.go]: Are these changes intended?

File: dp/cloud/python/magma/test_runner/tests/test_dp_with_orc8r.py [hunk 4]
```diff
@@ -327,6 +579,12 @@ def then_message_is_eventually_sent(self, filters: Dict[str, Any]):
         logs = self.when_logs_are_fetched(filters)
         self.assertEqual(logs["total_count"], 1)
 
+    def then_message_is_never_sent(self, filters: Dict[str, Any]):
+        for _ in range(15):
+            logs = self.when_logs_are_fetched(filters)
```
Comment (jkmar) [inline dp/cloud/python/magma/test_runner/tests/test_dp_with_orc8r.py]: This loop is unnecessary, since your are overwriting logs variable every time not doing anything else with it and also 15s seems way too long.

File: dp/cloud/python/magma/test_runner/tests/test_dp_with_orc8r.py [hunk 5]
```diff
@@ -261,10 +502,21 @@ def when_cbsd_is_deleted(self, cbsd_id: int):
...
         r = send_request_to_backend('put', f'cbsds/{cbsd_id}', json=data)
         self.assertEqual(r.status_code, expected_status)
 
+    def when_cbsd_is_updated_by_enodebd(self, req: EnodebdUpdateCbsdRequest):
+        resp = self.orc8r_dp_client.EnodebdUpdateCbsd(
+            request=req, metadata=(
+                (MAGMA_CLIENT_CERT_SERIAL_KEY, MAGMA_CLIENT_CERT_SERIAL_VALUE),
+            ),
+        )
+        self.assertEqual(b'', resp.SerializeToString())
```
Comment (jkmar) [inline dp/cloud/python/magma/test_runner/tests/test_dp_with_orc8r.py]: it would be better to check for errors instead

File: dp/cloud/python/magma/test_runner/tests/test_dp_with_orc8r.py [hunk 6]
```diff
@@ -225,25 +398,93 @@ def test_fetching_logs_with_custom_filters(self):
...
+        self.then_state_is(state, get_empty_state())
 
-        cbsd = self.when_cbsd_is_fetched(builder.serial_number)
-        self.then_cbsd_is(cbsd, builder.build_registered_active_data())
+        cbsd = self._check_cbsd_successfully_provisioned(builder)
 
         return cbsd['id']
 
+    def _check_cbsd_successfully_provisioned(self, builder: CbsdAPIDataBuilder) -> Dict[str, Any]:
+        sn = builder.payload["serial_number"]
```
Comment (jkmar) [inline dp/cloud/python/magma/test_runner/tests/test_dp_with_orc8r.py:455]: looks hacky - depending on internal builder structure

File: dp/cloud/python/magma/test_runner/tests/test_dp_with_orc8r.py [hunk 7]
```diff
@@ -225,25 +398,93 @@ def test_fetching_logs_with_custom_filters(self):
...
+            .with_grant()
+            .with_state("registered")
+            .with_cbsd_id(cbsd_id)
+            .payload,
+        )
+
+        return cbsd
+
+    def _check_provision_logs(self):
+        with self.while_cbsd_is_active():
```
Comment (jkmar) [inline dp/cloud/python/magma/test_runner/tests/test_dp_with_orc8r.py]: not sure about putting this here, since it would prevent someone from using it in the test, when this context manager was already called, for example:
```
def test_something(self):
    with self.while_cbsd_is_active():
        ....
        self._check_provision_logs() # it cannot be used here
```

File: dp/cloud/python/magma/test_runner/tests/integration_testcase.py [hunk 8]
```diff
@@ -25,5 +24,16 @@ def tearDownClass(cls) -> None:
...
 
+class Orc8rIntegrationTestCase(TestCase):
+    @classmethod
+    def setUpClass(cls) -> None:
+        super().setUpClass()
+
+        grpc_channel = grpc.insecure_channel(
+            f"{config.ORC8R_DP_GRPC_SERVICE}:{config.ORC8R_DP_GRPC_PORT}",
+        )
+        cls.orc8r_dp_client = CbsdManagementStub(grpc_channel)
```
Comment (jkmar) [inline dp/cloud/python/magma/test_runner/tests/integration_testcase.py:35]: It would be nice if we could setup metadata here, not to worry about it later in the tests

File: dp/cloud/python/magma/test_runner/tests/test_dp_with_orc8r.py [hunk 9]
```diff
@@ -39,46 +43,165 @@
...
+            .with_desired_state() \
+            .with_antenna_gain() \
+            .with_single_step_enabled(True) \
+            .with_serial_number(self.serial_number)
+
+        cbsd_id = self.given_single_step_cbsd_provisioned(builder)
+
+        self._check_provision_logs()
 
         self.delete_cbsd(cbsd_id)
```
Comment (jkmar) [inline dp/cloud/python/magma/test_runner/tests/test_dp_with_orc8r.py]: we can get rid of this delete (maybe leave it in just a one test)

File: dp/cloud/python/magma/test_runner/tests/test_dp_with_orc8r.py [hunk 10]
```diff
@@ -225,25 +398,93 @@ def test_fetching_logs_with_custom_filters(self):
...
+            .with_is_active(True)
+            .with_grant()
+            .with_state("registered")
+            .with_cbsd_id(cbsd_id)
+            .payload,
+        )
+
+        return cbsd
+
+    def _check_provision_logs(self):
```
Comment (jkmar) [inline dp/cloud/python/magma/test_runner/tests/test_dp_with_orc8r.py]: I'd rather call it `then...` to be consistent with `given, when, then` convention

File: dp/cloud/python/magma/test_runner/tests/test_dp_with_orc8r.py [hunk 11]
```diff
@@ -39,46 +43,165 @@
...
+                ),
+                cbsd_category="b",
+            )
+
+            self.when_cbsd_is_updated_by_enodebd(update_request)
+            cbsd = self.when_cbsd_is_fetched(self.serial_number)
+            self.then_cbsd_is(cbsd, builder.with_latitude_deg(100.1).with_longitude_deg(1000.1).payload)
+            self.then_message_is_eventually_sent(filters)
+
+    def test_cbsd_not_unregistered_when_coordinates_change_less_than_10_m(self):
```
Comment (jkmar) [inline dp/cloud/python/magma/test_runner/tests/test_dp_with_orc8r.py:135]: I'm actually wondering if we need to have integration tests for all such detailed scenarios (maybe UTs are enough)

File: dp/cloud/python/magma/test_runner/tests/api_data_builder.py [hunk 12]
```diff
@@ -0,0 +1,168 @@
...
+
+    def with_is_active(self, is_active: bool) -> CbsdAPIDataBuilder:
+        self.payload['is_active'] = is_active
+        return self
+
+    def with_single_step_enabled(self, enabled: bool) -> CbsdAPIDataBuilder:
+        self.payload['single_step_enabled'] = enabled
+        return self
+
+    def build_grant_state_data(self, frequenzy_mhz=None, bandwidth_mhz=None, max_eirp=None) -> CBSDStateResult:
```
Comment (jkmar) [inline dp/cloud/python/magma/test_runner/tests/api_data_builder.py:155]: What happened to the other build methods (this one is only for enodebd)? Do we no longer check GET requests?

File: dp/cloud/go/services/dp/storage/cbsd_manager.go (hunk 2 above)
Comment (WojSad) [reply dp/cloud/go/services/dp/storage/cbsd_manager.go:249]: it's in dp/storage/diff.go

File: dp/cloud/go/services/dp/storage/cbsd_manager.go (hunk 1 above)
Comment (WojSad) [reply dp/cloud/go/services/dp/storage/cbsd_manager.go]: done

File: orc8r/cloud/go/sqorc/error_checker_test.go (hunk 3 above)
Comment (WojSad) [reply orc8r/cloud/go/sqorc/error_checker_test.go]: I added a test case called "test nil error returns nil" and re-aligned the rest of them for better indentation

Comment (jkmar) [reply orc8r/cloud/go/sqorc/error_checker_test.go]: :+1:

File: dp/cloud/python/magma/test_runner/tests/test_dp_with_orc8r.py (hunk 4 above)
Comment (WojSad) [reply dp/cloud/python/magma/test_runner/tests/test_dp_with_orc8r.py]: lol, i must've been tired :)

File: dp/cloud/python/magma/test_runner/tests/test_dp_with_orc8r.py (hunk 5 above)
Comment (WojSad) [reply dp/cloud/python/magma/test_runner/tests/test_dp_with_orc8r.py]: done

File: dp/cloud/python/magma/test_runner/tests/integration_testcase.py (hunk 8 above)
Comment (WojSad) [reply dp/cloud/python/magma/test_runner/tests/integration_testcase.py:35]: leaving as is, as discussed

File: dp/cloud/python/magma/test_runner/tests/test_dp_with_orc8r.py (hunk 9 above)
Comment (WojSad) [reply dp/cloud/python/magma/test_runner/tests/test_dp_with_orc8r.py]: done

File: dp/cloud/python/magma/test_runner/tests/test_dp_with_orc8r.py (hunk 10 above)
Comment (WojSad) [reply dp/cloud/python/magma/test_runner/tests/test_dp_with_orc8r.py]: done

File: dp/cloud/python/magma/test_runner/tests/test_dp_with_orc8r.py (hunk 11 above)
Comment (WojSad) [reply dp/cloud/python/magma/test_runner/tests/test_dp_with_orc8r.py:135]: why not, the filter for coordinate diff may work properly, but for some reason the parent method may be giving it wrong info

Comment (jkmar) [reply dp/cloud/python/magma/test_runner/tests/test_dp_with_orc8r.py:135]: :+1:

**Review by jkmar — APPROVED** [review]
